services/MerchStoreService.ts

```diff
@@ -154,19 +173,24 @@ export default class MerchStoreService {
       const item = await merchItemRepository.findByUuid(uuid);
       if (!item) throw new NotFoundError();
       const hasBeenOrdered = await Repositories.merchOrderItem(txn).hasItemBeenOrdered(uuid);
-      if (hasBeenOrdered) throw new UserError('This item has been ordered already');
+      if (hasBeenOrdered) throw new UserError(UserErrors.MERCH_ITEM_ORDERED_FROM);
       return merchItemRepository.deleteMerchItem(item);
     });
   }
 
   public async createItemOption(item: Uuid, option: MerchItemOption): Promise<PublicMerchItemOption> {
     return this.transactions.readWrite(async (txn) => {
       const merchItem = await Repositories.merchStoreItem(txn).findByUuid(item);
-      if (!merchItem) throw new NotFoundError('Item not found');
+      if (!merchItem) throw new NotFoundError(NotFoundErrors.MERCH_ITEM);
+      if (!merchItem.hasVariants) throw new UserError(UserErrors.NO_ITEM_VARIANTS_ADD_OPTION);
+      const hasDifferentOptionType = merchItem.options && merchItem.options[0].metadata?.type !== option.metadata?.type;
```

Comment:
The portal should guarantee that all items have at least one option, so no need to check that `merchItem.options` exists and is not empty.